pkg/codegen/pcl/binder.go

```diff
@@ -46,6 +49,7 @@ type ComponentProgramBinderArgs struct {
 	BinderLoader                 schema.Loader
 	ComponentSource              string
 	ComponentNodeRange           hcl.Range
+	PackageCache                 *PackageCache
```

Comment:
Not sure we need this? Cache strikes me as internal state, not an argument? Is there a reason we have to add it here?

Reply:
This is specific to binding components. the thing is with component args is that they need to be able to reconstruct the binder options for binding the PCL files of that component. 

The reason we are sharing the cache between the parent binder and component (child) binder is so that we re-use bound parameterized packages from the main program in the component program without having to redefine the package block in component programs as well.